### PR TITLE
Nikola CMS: init at 7.8.3

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

Nikola is quite a popular and advanced python-based CMS, similar to Lektor.
It required a lot of updates to dependencies, and some new python dependencies, all of which are included.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by globin** [comment]
Could you move the dependencies to seperate files in `pkgs/development/python-modules/` and just call them from `python-packages.nix`. This will eventually happen with all the dependencies, the file is simply getting to large and this will make it easier to handle (git log, merge conflicts, etc.)

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -14595,6 +14651,36 @@ in {
...
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/N/${pname}/${name}.tar.gz";
+      sha256 = "12kcnvwy9dr194riji6iird1zpdcv252d19lfvabr39rsi9658jl";
+    };
+
+    meta = {
+      description = "A static content management system";
+      homepage    = "https://www.getnikola.com/";
+      license     = "MIT";
```
Comment (fpletz) [inline pkgs/top-level/python-packages.nix]: Please use `licenses.mit` here.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -6481,6 +6481,28 @@ in {
...
+    name = "${pname}-${version}";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/d/${pname}/${name}.tar.gz";
+      sha256 = "0ghj3bsrc5bm30w97bn8nvf5jlabg06xs16g7284mimyr7q9z0j0";
+    };
+
+    propagatedBuildInputs = with self; [ cloudpickle pyinotify pytest ];
+
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: include a reason why you disable them

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -6481,6 +6481,28 @@ in {
...
+    pname = "doit";
+    version= "0.30.0";
+    name = "${pname}-${version}";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/d/${pname}/${name}.tar.gz";
+      sha256 = "0ghj3bsrc5bm30w97bn8nvf5jlabg06xs16g7284mimyr7q9z0j0";
+    };
+
+    propagatedBuildInputs = with self; [ cloudpickle pyinotify pytest ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: does it need pytest during runtime??

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -12673,6 +12694,21 @@ in {
     };
   };
 
+  husl = buildPythonPackage rec {
+    pname    = "husl";
+    name    = "${pname}-${version}";
+    version = "4.0.3";
+
+    # ImportError: No module named tests
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: ok...and how come? Should the test runner be invoked differently, or are tests simply missing? Just stating there was an importerror is not sufficient.

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -14538,6 +14574,26 @@ in {
...
+    pname = "micawber";
+    version = "0.3.3";
+    name = "${pname}-${version}";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/m/${pname}/${name}.tar.gz";
+      sha256 = "18lna1bgzw4l3kqdi3nr1l557rj3wddrcj81x0rf8f4p87xsg9fq";
+    };
+
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: why?

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -14595,6 +14651,36 @@ in {
...
+  nikola = buildPythonPackage rec {
+    pname = "Nikola";
+    name = "${pname}-${version}";
+
+    version = "7.8.3";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/N/${pname}/${name}.tar.gz";
+      sha256 = "12kcnvwy9dr194riji6iird1zpdcv252d19lfvabr39rsi9658jl";
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: are tests working?

File: pkgs/top-level/python-packages.nix [hunk 7]
```diff
@@ -14625,6 +14711,25 @@ in {
     };
   };
 
+  phpserialize = self.buildPythonPackage rec {
+    pname = "phpserialize";
+    version = "1.3";
+    name = "${pname}-${version}";
+
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: why?

File: pkgs/top-level/python-packages.nix [hunk 8]
```diff
@@ -15298,6 +15403,25 @@ in {
     };
   };
 
+  piexif = self.buildPythonPackage rec {
+    pname = "piexif";
+    version = "1.0.10";
+    name = "${pname}-${version}";
+
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: why?

File: pkgs/top-level/python-packages.nix [hunk 9]
```diff
@@ -15315,16 +15439,15 @@ in {
   };
 
   pygal = buildPythonPackage rec {
-    version = "2.0.10";
-    name = "pygal-${version}";
+    pname = "pygal";
+    version = "2.3.1";
+    name = "${pname}-${version}";
 
     doCheck = !isPyPy;  # one check fails with pypy
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: if its only one test, and the test is broken, then surely we can just disable that one test?

**Review by FRidh — CHANGES_REQUESTED** [review]
Are you going to maintain these Python packages?

**Comment by FRidh** [comment]
status update?

**Pull request closed without merging**